### PR TITLE
Getitemnd

### PR DESCRIPTION
**Pull request by jriehl**
Adds support for 1d indexing into ndarrays from no-Python mode.

### CODE REVIEW

**Comment by seibert** [comment]
Can the partially indexed ndarray be passed to another nopython function?  It would be good to demonstrate that with an additional unit test.

Also, this PR should include the documentation update to numpysupported.rst to indicate that partial indexing is allowed.

File: numba/targets/arrayobj.py [hunk 1]
```diff
@@ -114,18 +114,36 @@ def iternext_array(context, builder, sig, args, result):
...
+        result = _getitem_array1d(context, builder, aryty, adapted_ary, idx,
+                                  wraparound=idxty.signed)
+    elif ndim > 1:
+        out_ary_ty = make_array(aryty.copy(ndim = ndim - 1))
+        out_ary = out_ary_ty(context, builder)
+        in_shapes = cgutils.unpack_tuple(builder, adapted_ary.shape, count=ndim)
+        in_strides = cgutils.unpack_tuple(builder, adapted_ary.strides,
+                                          count=ndim)
+        out_ary.parent = adapted_ary.parent
+        out_ary.nitems = builder.udiv(adapted_ary.nitems, in_shapes[0])
```
Comment (pitrou) [inline numba/targets/arrayobj.py]: How does it work with 0-element arrays, e.g. `np.array([]).reshape(0, 1, 0)`?

Comment (jriehl) [inline numba/targets/arrayobj.py]: Not very well, I presume.  What would you suggest?

Comment (pitrou) [inline numba/targets/arrayobj.py]: Just force `out_ary.nitems` to 0 in case `adapted_ary.nitems` is 0?

Comment (jriehl) [inline numba/targets/arrayobj.py]: Makes sense to me.  Will do.

**Pull request merged**